### PR TITLE
Fix/update flask

### PR DESCRIPTION
**Pull request by george42-ctds**
Jira Ticket: [PPS-629](https://ctds-planx.atlassian.net/browse/PPS-629)

### Dependency updates

* Upgrade flask

### Deployment changes

[PPS-629]: https://ctds-planx.atlassian.net/browse/PPS-629?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: bin/settings.py [hunk 1]
```diff
@@ -12,6 +12,8 @@ def load_json(file_name):
 conf_data = load_json("creds.json")
 config = app.config
 
+config["SESSION_REFRESH_EACH_REQUEST"] = False
```
Comment (paulineribeyre) [inline bin/settings.py:15]: what is this for?

**Comment by paulineribeyre** [comment]
Sheepdog/Indexd have not been updated yet to a more recent version of Flask but this needs to be merged. Temporarily removes Sheepdog from the dev dependencies. The unit tests fail as expected, but I think the integration tests cover enough of the peregrine/sheepdog interaction (will be fixed in PPS-655)

Also ignoring Snyk request to upgrade pyyaml for now (will be fixed in PXP-11163)

File: bin/settings.py (hunk 1 above)
Comment (george42-ctds) [reply bin/settings.py:15]: This config setting was mentioned in the Overview section of the ticket. After re-reading the Overview, I think this is unnecessary after the upgrade in Flask.

Comment (paulineribeyre) [reply bin/settings.py:15]: ok, i'll remove it

**Pull request merged**